frontend/src/main/java/org/abs_models/frontend/typechecker/ext/TypeExtensionHelper.java

```diff
@@ -116,6 +117,7 @@ public void checkAssignStmt(AssignStmt s) {
         }
 
         checkAssignable(s.getValue().getType(),s.getVar().getType(), s);
+        checkAssignable(s.getVar().getType(), s.getValue(), s);
```

Comment:
Hmm, we have two `checkAssignable` methods with different signatures; one or both should be renamed so we can see what they do.

Alternatively, can the two methods be merged into one, in case they always go together?  (Are there cases where we would call one but not the other?)

Reply:
I have renamed the one with the expression to `checkAssignableBehaviorType` to make the distinction clear(er). I think there are scenarios where you only need one of the methods and I would prefer to leave it open to future extensions, how to best combine behavior and data types.